src/aggregator/aggregator/flush.go

```diff
@@ -38,6 +38,11 @@ type flushingMetricList interface {
 	// FlushInterval returns the periodic flush interval.
 	FlushInterval() time.Duration
 
+	// FixedFlushOffset returns a fixed flush offset (and true) within the flush interval when
+	// a flush is performed for lists that have a fixed flush offset.
+	// For lists that do not have a fixed flush offset, it returns 0 and false.
+	FixedFlushOffset() (time.Duration, bool)
```

Comment:
Dropping `fixedOffsetFlushingMetricList` interface and moving the method here (with `ok` flag) to avoid the error-prone type check in https://github.com/m3db/m3/pull/3292/files#diff-5a6baf266bd455b257e1a2f2ded077f3f2373a44628318748d2008d9ba59b9bcL284 which was masking the fact that `timedMetricList` was not exposing its `flushOffset`.